### ISSUE TITLE
Vivado IPI compatibility

### ISSUE THREAD

**Issue by iljabek** [tgingold-cern/cheby#42]
if using the cheby-generated AXI4-Lite interface in Vivado block design without a wrapper following issues arise:

* the naming of the interface does not follow usual scheme of `[MS]_AXI_NAME`, instead `slave` is produced
* even if enabling the x-attributes, the `reset` pin does not quite follow auto-discovery naming scheme because of underscore in `areset_n` and is not associated with the `aclk` clock pin
* the `aclk` pin is not automatically associated with the port in properties

### minimal yaml example:

```yaml
memory-map:
  name: info
  bus: axi4-lite-32
  x-hdl:
    bus-attribute: Xilinx
```

### Current IP in BD + post-autoconnect:

![cheby_default_clk2axi](https://github.com/tgingold-cern/cheby/assets/12845618/28940f31-809f-49b8-a4d7-8b367ae195e0)
![cheby_default_ipi](https://github.com/tgingold-cern/cheby/assets/12845618/a0f8ad03-3497-4c38-a23c-bcab9908bc77)
![cheby_default_ipi_conn](https://github.com/tgingold-cern/cheby/assets/12845618/cdf8c574-1ec1-4642-8317-57a35ad3e7f4)

### minimal "valid" IP in BD  + post-autoconnect:
![cheby_xattr_clk2axi](https://github.com/tgingold-cern/cheby/assets/12845618/4201034f-baf1-4dd8-b67b-489c03649fbd)
![cheby_xattr_ipi_conn](https://github.com/tgingold-cern/cheby/assets/12845618/2345b349-283c-4f8a-90b4-0a9632566632)

* this can be achieved via x-attributes
* desirable is also to customize the names:

![image](https://github.com/tgingold-cern/cheby/assets/12845618/009af109-e758-4e59-ba22-aa3e25f26557)

**Comment by iljabek** [comment]
@tgingold-cern I have to look into the failing test, but is this https://github.com/iljabek/cheby/commit/095f911874880a79eeb2a5248dca565f75e972bb going in good general direction? would you prefer to split the minimal and customization commits?

**Comment by tgingold-cern** [comment]
Yes, I think this is OK.  I would use 'clock-name' and 'reset-name' instead of 'bus-clock-name' and 'bus-reset-name'

Not sure why you need 'use-bus-name-in-ports'.  There is a 'reg-prefix' which might do what you'd like.

**Comment by iljabek** [comment]
> There is a 'reg-prefix' which might do what you'd like

can the bus name consist only out of `reg-prefix`, i.e. `name` be empty?
